test_runner/fixtures/zenith_fixtures.py

```diff
@@ -126,29 +128,28 @@ def pageserver(zenith_cli):
 class Postgres:
     """ An object representing a running postgres daemon. """
 
-    def __init__(self, zenith_cli, repo_dir):
+    def __init__(self, zenith_cli, repo_dir, instance_num):
         self.zenith_cli = zenith_cli
+        self.instance_num = instance_num
         self.running = False
         self.host = 'localhost'
-        self.port = 55432
+        self.port = 55431 + instance_num
         self.repo_dir = repo_dir
-        # path to conf is <repo_dir>/pgdatadirs/pg1/postgresql.conf
+        # path to conf is <repo_dir>/pgdatadirs/pg<instance_num>/postgresql.conf
 
-    def create_start(self, config_lines=None):
+    def create_start(self, branch="main", config_lines=None):
         """ create the pg data directory, and start the server """
-        self.zenith_cli.run(['pg', 'create'])
+        self.zenith_cli.run(['pg', 'create', branch])
         if config_lines is None:
             config_lines = []
         self.config(config_lines)
-        # FIXME: where did the name pg1 come from?
-        self.zenith_cli.run(['pg', 'start', 'pg1'])
+        self.zenith_cli.run(['pg', 'start', 'pg{}'.format(self.instance_num)])
         self.running = True
-
+        return
 
     #lines should be an array of valid postgresql.conf rows
     def config(self, lines):
-        #TODO use real node name, not just guessed pg1
-        filename = 'pgdatadirs/pg1/postgresql.conf'
+        filename = 'pgdatadirs/pg{}/postgresql.conf'.format(self.instance_num)
```

Comment:
I wish we didn't have to guess at the name "pgX" that was created.

Should we get this as feedback from the CLI? Should we specify the name we want?

Reply:
And by "I wish" I don't mean "should fix now". I just meant "at some point in the future".